src/SonarScanner.MSBuild.Common/MsBuildPathSettings.cs

```diff
@@ -21,69 +21,155 @@
+            return new[] { GetMsBuildImportBeforePath(userProfilePath, "15.0") }; // Older versions are not supported on non-Windows OS
+        }
+
+        /// <summary>
+        /// Returns the local AppData path for the current user. This method will return multiple paths if running as Local System.
+        /// </summary>
+        private IEnumerable<string> GetLocalApplicationDataPaths()
+        {
+            var localAppData = environmentGetFolderPath(
+                Environment.SpecialFolder.LocalApplicationData,
+                Environment.SpecialFolderOption.Create);
+
+            // Return empty enumerable when Local AppData is empty. In this case an exception should be thrown at the call site.
+            if (string.IsNullOrWhiteSpace(localAppData))
+            {
+                yield break;
+            }
+
+            yield return localAppData;
 
-            if (!this.isWindows())
+            // The code below is Windows-specific, no need to be executed on non-Windows platforms.
+            if (!isWindows())
             {
-                var userProfilePath = this.getEnvironmentSpecialFolderPath(Environment.SpecialFolder.UserProfile, true);
+                yield break;
+            }
+
+            // When running under Local System account on a 64bit OS, the local application data folder
+            // is inside %windir%\system32
+            // When a process copies a file in this location, the OS will automatically redirect it to:
+            // for 32bit processes - %windir%\sysWOW64\...
+            // for 64bit processes - %windir%\system32\...
+            // Nice explanation could be found here:
+            // https://www.howtogeek.com/326509/whats-the-difference-between-the-system32-and-syswow64-folders-in-windows/
+            // If a 32bit process needs to copy files to %windir%\system32, it should use %windir%\Sysnative
+            // to avoid the redirection:
+            // https://docs.microsoft.com/en-us/windows/desktop/WinProg64/file-system-redirector
+            // We need to copy the ImportBefore.targets in both locations to ensure that both the 32bit and 64bit versions
+            // of MSBuild will be able to pick them up.
+            var systemPath = environmentGetFolderPath(
+                Environment.SpecialFolder.System,
+                Environment.SpecialFolderOption.None); // %windir%\System32
+            if (!string.IsNullOrWhiteSpace(systemPath) &&
+                localAppData.StartsWith(systemPath)) // We are under %windir%\System32 => we are running as System Account
+            {
+                var systemX86Path = environmentGetFolderPath(
+                    Environment.SpecialFolder.SystemX86,
+                    Environment.SpecialFolderOption.None); // %windir%\SysWOW64 (or System32 on 32bit windows)
+                var localAppDataX86 = localAppData.ReplaceCaseInsensitive(systemPath, systemX86Path);
```

Comment:
Changed to case-insensitive replacement because this code is executed on Windows (it should not matter, because the paths are coming from the OS, but just in case)